starknet_py/net/full_node_client.py

```diff
@@ -849,74 +867,100 @@ def _create_broadcasted_declare_v2_properties(transaction: DeclareV2) -> dict:
     return declare_v2_properties
 
 
-def _create_broadcasted_invoke_properties(transaction: Invoke) -> dict:
+def _create_broadcasted_declare_v3_properties(transaction: DeclareV3) -> dict:
+    contract_class = cast(
+        Dict, SierraCompiledContractSchema().dump(obj=transaction.contract_class)
+    )
+
+    declare_v3_properties = {
+        "contract_class": {
+            "entry_points_by_type": contract_class["entry_points_by_type"],
+            "sierra_program": contract_class["sierra_program"],
+            "contract_class_version": contract_class["contract_class_version"],
+        },
+        "sender_address": _to_rpc_felt(transaction.sender_address),
+        "compiled_class_hash": _to_rpc_felt(transaction.compiled_class_hash),
+        "account_deployment_data": [
+            _to_rpc_felt(data) for data in transaction.account_deployment_data
+        ],
+    }
+
+    if contract_class["abi"] is not None:
+        declare_v3_properties["contract_class"]["abi"] = contract_class["abi"]
+
+    return {
+        **_create_broadcasted_txn_v3_common_properties(transaction),
+        **declare_v3_properties,
+    }
+
+
+def _create_broadcasted_invoke_properties(transaction: Union[Invoke, InvokeV3]) -> dict:
     invoke_properties = {
         "sender_address": _to_rpc_felt(transaction.sender_address),
         "calldata": [_to_rpc_felt(data) for data in transaction.calldata],
     }
+
+    if isinstance(transaction, InvokeV3):
+        return {
+            **_create_broadcasted_txn_v3_common_properties(transaction),
+            **invoke_properties,
+            "account_deployment_data": [
+                _to_rpc_felt(data) for data in transaction.account_deployment_data
+            ],
+        }
+
     return invoke_properties
 
 
-def _create_broadcasted_deploy_account_properties(transaction: DeployAccount) -> dict:
+def _create_broadcasted_deploy_account_properties(
+    transaction: Union[DeployAccount, DeployAccountV3]
+) -> dict:
     deploy_account_txn_properties = {
         "contract_address_salt": _to_rpc_felt(transaction.contract_address_salt),
         "constructor_calldata": [
             _to_rpc_felt(data) for data in transaction.constructor_calldata
         ],
         "class_hash": _to_rpc_felt(transaction.class_hash),
     }
+
+    if isinstance(transaction, DeployAccountV3):
+        return {
+            **_create_broadcasted_txn_v3_common_properties(transaction),
+            **deploy_account_txn_properties,
+        }
+
```

Comment:
Nit: From future refactoring pov, I think using different approaches in all three functions that do a similar thing is a bit sub-optimal.